pysteps/tests/test_motion.py

```diff
@@ -324,6 +325,35 @@ def test_input_shape_checks(
             motion_method(np.zeros((frames, image_size, image_size)), verbose=False)
 
 
+def test_vet_padding():
+    """
+    Test that the padding functionality in vet works correctly with ndarrays and
+    masked arrays.
+    """
+
+    _, precip_obs = _create_observations(
+        reference_field.copy(), "linear_y", num_times=2
+    )
+
+    precip_obs = precip_obs[
+        :, 200:427, 250:456
+    ]  # Use a small region to speed up the test
+
+    # precip_obs.shape == (227 , 206)
+    # 227 is a prime number ; 206 = 2*103
+    vet_method = partial(
+        motion.get_method("vet"),
+        verbose=False,
+        sectors=((16, 4, 2), (16, 4, 2)),
+        options=dict(maxiter=5)
+        # We use only a few iterations since
+        # we don't care about convergence in this test
+    )
+
+    assert precip_obs.shape == vet_method(precip_obs).shape
+    assert precip_obs.shape == vet_method(np.ma.masked_invalid(precip_obs)).shape
```

Comment:
not sure here, isn't padding = 0 by default?

Reply:
Oops, the test was not testing the padding code :confounded:
Commits 1dc4eee and 35ca32a add the tests and fix a bug with the mask padding.